DataFormats/L1TrackTrigger/interface/TTDTC.h

```diff
@@ -0,0 +1,74 @@
+#ifndef __L1TrackTrigger_TTDTC__
+#define __L1TrackTrigger_TTDTC__
+
+#include "DataFormats/L1TrackTrigger/interface/TTTypes.h"
+#include "DataFormats/L1TrackTrigger/interface/TTBV.h"
+
+#include <bitset>
+#include <utility>
+#include <vector>
+
+/*! 
+ * \class  TTDTC
+ * \brief  Class to store hardware like structured TTStub Collection used by Track Trigger emulators
+ * \author Thomas Schuh
+ * \date   2020, Jan
+ */
+class TTDTC {
+public:
+  // outer tracker DTC types
+  typedef std::bitset<TTBV::S> BV;         // bit accurate Stub
+  typedef std::pair<TTStubRef, BV> Frame;  // TTStub with bit accurate Stub
+  typedef std::vector<Frame> Stream;       // stub collection transported over an optical link between DTC and TFP
+  typedef std::vector<Stream> Streams;     // collection of optical links
+
+public:
+  TTDTC() {}
+
+  TTDTC(const int& numRegions, const int& numOverlappingRegions, const int& numDTCsPerRegion);
+
+  ~TTDTC() {}
+
+  // Access to product configurations
+  int numRegions() const;
+  int numDTCBoards() const;
+  int numDTCChannel() const;
+  int numTFPChannel() const;
+
+  // Access to one specific stream of TTStubRefs using DTC identifier (region[0-8], board[0-23], channel[0-1])
+  // dtcRegions aka detector regions are defined by tk layout
+  Stream& dtcStream(const int& dtcRegion, const int& dtcBoard, const int& dtcChannel);
+
+  // all TFP identifier (region[0-8], channel[0-47])
+  std::vector<int> tfpRegions() const;
+  std::vector<int> tfpChannels() const;
+
+  // Access to one specific stream of TTStubRefs using TFP identifier (region[0-8], channel[0-47])
+  // tfpRegions aka processing regions are rotated by -0.5 region width w.r.t detector regions
+  Stream tfpStream(const int& tfpRegion, const int& tfpChannel) const;
```

Comment:
Return a `const` reference? Also the parameters are more performant with just `int` instead of references (also below).